flocker/control/httpapi.py

```diff
@@ -185,6 +185,63 @@ def saved(ignored):
         saving.addCallback(saved)
         return saving
 
+    # Will we also have a /state endpoint? Or just route directly to the datasets path?
+    @app.route("/state/datasets", methods=['GET'])
+    @user_documentation("""
+        Get current cluster datasets
+        """, examples=[u"get state datasets"])
+    @structured(
+        inputSchema={},
+        outputSchema={'$ref': '/v1/endpoints.json#/definitions/datasets_array'},
+        schema_store=SCHEMAS
+    )
+    def datasets(self):
+        """
+        Return the current primary datasets in the cluster.
+        """
+        deployment = self.cluster_state_service.as_deployment()
+        return list(datasets_from_deployment(deployment))
+
+
+def datasets_from_deployment(deployment):
+    """
+    Extract the primary datasets from the supplied deployment instance.
+
+    XXX: What would be the point in returning secondary dataset dictionaries
+    here? They'd be exactly the same as the primary...unless the metadata or
```

Comment:
For the ZFS backend, they will probably eventually include extra information: something reflecting how up-to-date they are with respect to the primary.  Their existence at all may be useful information as well - for example, to inform cleanup decisions.